File: regexlib/javascript_test_file/regexlib_4902.js
/* 4902
 * ^(([a-z])+.)+[A-Z]([a-z])+$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"a"*64+"!1 __EOA(iv)"
 */
var REGEX = /^(([a-z])+.)+[A-Z]([a-z])+$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'a'.repeat(i*1) + '!1 __EOA(iv)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_4902.py
# 4902
# ^(([a-z])+.)+[A-Z]([a-z])+$
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"a"*64+"!1 __EOA(iv)"

import re2 as re
from time import perf_counter

regex = """^(([a-z])+.)+[A-Z]([a-z])+$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "a" * i * 1 + "!1 __EOA(iv)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")